src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -769,12 +769,21 @@ protected override TypeWithAnnotations InferTypeOfVarVariable(DiagnosticBag diag
                         var initializer = (ConstructorInitializerSyntax)_nodeToBind;
                         _nodeBinder.BindConstructorInitializer(initializer, diagnostics);
                         break;
-                    case SyntaxKind.SimpleBaseType:
-                        _nodeBinder.BindConstructorInitializer((SimpleBaseTypeSyntax)_nodeToBind, diagnostics);
+                    case SyntaxKind.PrimaryConstructorBaseType:
+                        _nodeBinder.BindConstructorInitializer((PrimaryConstructorBaseTypeSyntax)_nodeToBind, diagnostics);
                         break;
                     case SyntaxKind.ArgumentList:
-                        var invocation = (ConstructorInitializerSyntax)_nodeToBind.Parent;
-                        _nodeBinder.BindConstructorInitializer(invocation, diagnostics);
+                        switch (_nodeToBind.Parent)
+                        {
+                            case ConstructorInitializerSyntax ctorInitializer:
+                                _nodeBinder.BindConstructorInitializer(ctorInitializer, diagnostics);
+                                break;
+                            case PrimaryConstructorBaseTypeSyntax ctorInitializer:
+                                _nodeBinder.BindConstructorInitializer(ctorInitializer, diagnostics);
```

Comment:
>BindConstructorInitializer [](start = 44, length = 26)

Can you please identify a test for this code path?

Reply:
> Can you please identify a test for this code path?

For example, sBaseArguments_19, ``` OutVarTests.VerifyModelForOutVar(speculativeModel, xxDecl, xxRef);```

---
In reply to: [443929533](https://github.com/dotnet/roslyn/pull/45351#discussion_r443929533) [](ancestors = 443929533)